### PR TITLE
[Solvers] Settlements Table

### LINKED ISSUE #234: [Solvers] Settlements Table
As part of #156, we need to create  the Active solvers table. This will include: 

- Settlements Table.
- Logic to fetch settlements from the subgraph using the sdk.

Screen Shot 2022-10-03 at 21 29 54

Figma designs: https://www.figma.com/file/Y7pY6gGH2Wss6qhJwVcDJk/CoW-Explorer-(migrated)?node-id=0%3A1

### PR DESCRIPTION
**Pull request by matextrem**
# Summary

Closes #234 

Added Settlements table in Solvers page

Screen Shot 2022-10-05 at 20 19 02

# To Test

1.  Open the page `solvers` . You can use the header menu to navigate.
2. Click on Settlements tab
    * You'll see the settlements table.

**Notes:** For ETH cost, we need some Subgraph PR to be merged first before we can get that info into the table.

### CODE REVIEW

**Review by ramirotw — CHANGES_REQUESTED** [review]
Looking good.

A few things:

1. Change the base branch to `221/active-solvers` so the commits are cleaned
2. Lets hide for now the `ETH cost` column as we don't have that value from the subgraph
3. The `Tx hash` link should go to the `Transaction details` page not Etherscan

**Comment by ramirotw** [comment]
> Just want to clarify it with you whether we really need to hide it as it was a request from Anxo in [#156 (comment)](https://github.com/cowprotocol/explorer/issues/156#issuecomment-1184754982) (case 3). Or it is not going to be merged to Subgraph in the near future?

@GabrielCamba added a `txCostNative` field in the Settlement entity but currently that field is not the tx cost in eth (gas price * gas used) but just the gas price, so he needs to update that and re-deploy. Also, that field is only in the staging environment, you can it check [here](https://api.thegraph.com/subgraphs/name/cowprotocol/cow-staging/graphql?query=%7B%0A++settlements%28%0A++++where%3A+%7BtxHash%3A+%220x3356df15aa209e59e35786cdefdffbfde73eb2de3ec617fb0864291e773395a5%22%7D%0A++%29+%7B%0A++++firstTradeTimestamp%0A++++txHash%0A++++txCostNative%0A++++txCostUsd%0A++++solver+%7B%0A++++++address%0A++++%7D%0A++++trades+%7B%0A++++++sellToken+%7B%0A++++++++address%0A++++++++symbol%0A++++++%7D%0A++++++buyToken+%7B%0A++++++++address%0A++++++++symbol%0A++++++%7D%0A++++%7D%0A++%7D%0A%7D).

**Comment by elena-zh** [comment]
Hey @matextrem , nice changes! 
I will list here what is left, and new issues I have found: 
1. Still, navigation to TX details is Explorer is not fixed (should be in the same tab)
not fixed
2. I still see only 100 Settlements in GC and Mainnet
3. When hover a mouse on a native token (ETH/XDAI), 'Unknown' is displayed in the tooltip
unknow token looks like ETH
xdai
4. '+1' may be displayed when there are many tokens in a settlement. I would show what tokens are included into '+1' in the tooltip instead of nothing
1 - show which ones in tooltip
5. Could we make token icons look nicer? I see some kind of curved boarders on tokens, some kind of a white border or so
boarders

Thanks!

**Comment by matextrem** [comment]
Regarding the first comment, the subgraph is responding with empty symbol for MKR token. @ramirotw any suggestions?

File: src/apps/explorer/pages/Solver/index.tsx [hunk 1]
```diff
@@ -95,40 +109,15 @@ const Solver: React.FC = () => {
...
-    if (query.length) {
-      setPageOffset(0)
-    }
-  }, [query, setPageOffset])
-
-  useEffect(() => {
-    setQuery('')
-    setPageOffset(0)
-  }, [networkId, setPageOffset, setQuery])
+  }, [networkId, setPageOffset, setQuery, tabViewSelected])
```
Comment (alfetopito) [inline src/apps/explorer/pages/Solver/index.tsx:112]: Is the useEffect intentionally denpendent on `tabViewSelected` although it's not used in it?

**Review by alfetopito — COMMENTED** [review]
The query to fetch dune info is not used in the first tab, resulting in missing solver names, where on the second they are filled

Screen Shot 2022-10-18 at 17 34 35
Screen Shot 2022-10-18 at 17 34 10

File: src/apps/explorer/pages/Solver/index.tsx (hunk 1 above)
Comment (matextrem) [reply src/apps/explorer/pages/Solver/index.tsx:112]: This is intentional since we need to reset the search bar query  when changing between tabs

**Review by alfetopito — APPROVED** [review]
Looking great Mati!

Only one last thing that I noticed now, but I rather it be addressed in another PR as this one is already quite big.

The table is "jumping" when I move between tabs, likely because it triggers a reload on every change:

https://user-images.githubusercontent.com/43217/196759390-efa01e20-ac75-41b6-8e46-7ae6a71f13d3.mov

So I'm in favor of merging and addressing the pending stuff in different PRs

**Comment by elena-zh** [comment]
Hey @matextrem , 

changes LGTM besides know issues related to the Graph: 
 - MKR token is shown as 'unknown' token for the Mainnet
 - the amount of settlements is 1000 in GC and Mainnet.
 - total settlements is 100 on the Active solvers tab

Besides, I have created separate issues to address them later: 
 - Show a better message for Solvers tables when Graph API request fails/user is offline #239 
 - Navigate to the correct page selection when go back to the Settlements page from Tx details #238
 - Solvers: The table is "jumping" when move between tabs #240

The only one issues that I would like you to fix is that some tokens (GNO/MIR) are too dark on the table, while the icons look good on Tx details pages: 

gno too dark
icon
mir
mir1

Could you please fix it? 

Thanks!

**Comment by alfetopito** [comment]
Specifically regarding this point

> MKR token is shown as 'unknown' token for the Mainnet
> Regarding the first comment, the subgraph is responding with empty symbol for MKR token

I suspect I know the reason.

Maker and and another token (I think DAI?) have the `name` stored as `bytes32` rather than as string, which can be the root cause for this.

When fetching the token name from the contract, we are querying both versions of the contract to be safe.

Since this empty string is coming from the Subgraph, I suggest to:

1. Try to look it up in the local list of tokens - maybe it's already loaded
2. Follow the same loading step as other pages - try to get it from the contract

**Review by ramirotw — CHANGES_REQUESTED** [review]
The `Search by solver, address or name` placeholder in the table header should be updated. Right now it's filtering by solver name and tx hash, it would also be helpful to filter by solver address so I don't need to go to the Active solvers tab to know the name based on the address and then come back and use that name to get the latest settlements by solver.

**Comment by elena-zh** [comment]
Hey @matextrem , icons LGTM!
As for the MKR token, it is still unknown
mkr

Also, AFAIU from @ramirotw comment, it would be great to have a possibility to search by a solver address in the Settlements table, but you removed this option from the placeholder and, of cause, the search by a solver address is not working there.

**Comment by matextrem** [comment]
@elena-zh updated! 

MKR problem solved and search by address is working now. 

Screen Shot 2022-10-21 at 14 44 27

**Comment by elena-zh** [comment]
Hey @matextrem , changes LGTM!
The last nitpick is to change the placeholder in the search field to 'Search by tx hash, solver name or address'
image

Thanks

**Comment by matextrem** [comment]
@elena-zh Updated!

**Review by ramirotw — APPROVED** [review]
👌

**Pull request merged**